### DISCUSSION TITLE
Is there a good v2 Resource?

### DISCUSSION THREAD

**Discussion by Farmers-Tyler-Dougherty** [neutralinojs/neutralinojs#405] [Q&A]
I just found this lib and have been testing it out as a replacement for Electron.  I've been able to use the minimal template to get it launched (window / browser) on my Windows 10 machine.  I cannot get ANY of the built-in Neutralino API methods to work and have yet to be able to find any decent docs on how it should be working.  I've dug around in the lib's JS file and it seems like the methods do exist still, though none seem to do anything.  The Neutralino global vars are working correctly, just nothing else in the API.

In this example, neither the "onSucces" nor the "onError" seem to fire.  Is it recommended to use version 1 at this point?

``` JavaScript
Neutralino.os.getEnvar('USERNAME', 
    (data) => {
        document.getElementById('winver').innerHTML = data.value;
    },
    () => {
        console.log('error');
    }
);
```

**Comment by Farmers-Tyler-Dougherty** [comment]
[Just found this which seems to be current for the v2 API](https://github.com/neutralinojs/neutralino.js/tree/5c1bf4d897d383411bc2300762130cc6fcd285ec/src/api). 
I have begun refactoring my code to match this and have had some success.

``` JavaScript
Neutralino.os.getEnvar({
    key: 'OS',
    onSuccess: (data) => {
        document.getElementById('winver').innerText =  " " + data.value;
    },
    onError: (d) => {
        console.log(d);
    },
});
```

**Reply by shalithasuranga** [reply]
Great that you found the structure of the input parameters and thanks for trying Neutralinojs. We are in the process of upgrading the documentation for v2. We will do it soon as possible.

**Reply by shalithasuranga** [reply]
Hey.. Now the documentation for v2 is ready. Please check: https://neutralino.js.org/docs/
